paddle/framework/operator.h

```diff
@@ -84,7 +71,8 @@ class OperatorBase {
 
   /// InferShape infer the size of Variables used by this Operator with
   /// information inside scope
-  virtual void InferShape(const std::shared_ptr<Scope>& scope) const = 0;
+  virtual void InferShape(const std::shared_ptr<Scope>& scope) const final;
```

Comment:
Here we use the reference of `std::shared_ptr<Scope>` as a parameter, and all `InferShape` share the same `std::shared_ptr<Scope>`.
Does it means that there will be only one `std::shared_ptr<Scope>` ? If so, why not use `std::unique_ptr<Scope>` instead?